search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -33,9 +36,109 @@ class TemporalEntityHandler(
     private val temporalEntityAttributeService: TemporalEntityAttributeService,
     private val queryService: QueryService,
     private val authorizationService: AuthorizationService,
-    private val applicationProperties: ApplicationProperties
+    private val applicationProperties: ApplicationProperties,
+    private val entityEventService: EntityEventService
 ) {
 
+    /**
+     * Implements 6.18.3.1 - Create Entity or Update Temporal Representation of Entity
+     */
+    @PostMapping(consumes = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    suspend fun create(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @RequestBody requestBody: Mono<String>
+    ): ResponseEntity<*> {
+        val sub = getSubFromSecurityContext()
+        val body = requestBody.awaitFirst().deserializeAsMap()
+        val contexts = checkAndGetContext(httpHeaders, body)
+
+        return either<APIException, ResponseEntity<*>> {
```

Comment:
wondering about two things:
- this is a large method, better have pieces extracted out
- could this be simplified / more fluent?